autogen/src/main.rs

```diff
@@ -61,65 +62,82 @@ fn main() {
         panic!("SPIRV-Headers missing - please checkout using git submodule");
     }
 
-    let mut contents = String::new();
-
-    {
-        let path = autogen_src_dir
-            .join("external/SPIRV-Headers/include/spirv/unified1/spirv.core.grammar.json");
-        let mut file = fs::File::open(path).unwrap();
-        file.read_to_string(&mut contents).unwrap();
-    }
-
     let grammar: structs::Grammar = {
-        let mut original = serde_json::from_str(&contents).unwrap();
+        let mut original =
+            serde_json::from_str(
+                &std::str::from_utf8(
+                    &fs::read(autogen_src_dir.join(
+                        "external/SPIRV-Headers/include/spirv/unified1/spirv.core.grammar.json",
+                    ))
+                    .unwrap(),
+                )
+                .unwrap(),
+            )
+            .unwrap();
```

Comment:
```suggestion
        let mut original = serde_json::from_str(
            &fs::read_to_string(
                autogen_src_dir
                    .join("external/SPIRV-Headers/include/spirv/unified1/spirv.core.grammar.json"),
            )
            .unwrap(),
        )
        .unwrap();
```

Maybe keep this simple with `read_to_string`? Same below.

Reply:
Yes! I didn't know about it ><.
There really should be a note in std:fs::read about read_to_string existence. It's not the first time this annoys me ^^